src/Worker.Extensions.DurableTask/Execution/DurableFunctionExecutor.Activity.cs

```diff
@@ -36,7 +37,8 @@ private async ValueTask RunActivityAsync(FunctionContext context, BindingMetadat
         }
         catch (Exception ex)
         {
-            throw new DurableSerializationException(ex);
+            IExceptionPropertiesProvider? exceptionPropertiesProvider = context.InstanceServices.GetService(typeof(IExceptionPropertiesProvider)) as IExceptionPropertiesProvider;
```

Comment:
Can you inject this into the class instead of fetching it?

If not, there is a generic method you could use:

```suggestion
            IExceptionPropertiesProvider? exceptionPropertiesProvider = context.InstanceServices.GetService<IExceptionPropertiesProvider>();
```

Reply:
I missed the DI that happens in the main portion of the partial class. Updated it there as I prefer that over the service fetch.